tensorflow/python/ops/special_math_ops.py

```
@@ -151,70 +152,76 @@ def einsum(equation, *inputs):
         indices in its subscript, or
       - the input shapes are inconsistent along a particular axis.
   """
-  if '...' in equation:
-    raise ValueError('Subscripts with ellipses are not yet supported.')
-
-  match = re.match('([a-z,]+)(->[a-z]*)?', equation)
-  if not match:
-    raise ValueError(
-        'Indices have incorrect format: %s' % equation
-    )
-
-  inputs = list(inputs)
-  input_axis_labels = match.group(1).split(',')
-
-  if len(inputs) != len(input_axis_labels):
-    raise ValueError('Got %d arguments for equation "%s", expecting %d' % (
-        len(inputs), equation, len(input_axis_labels)))
+  name = kwargs.pop("name", None)
+  if kwargs:
+    key = list(kwargs.keys())[0]
+    raise TypeError("'{}' is an invalid keyword argument "
```

Should we raise an error for all the keys passed, not just the first one?